### ISSUE TITLE
"All Time" (Any date) cause UX issues

### ISSUE THREAD

**Issue by Dunge** [dangrossman/bootstrap-daterangepicker#1592]
Not a bug, but I'm looking at suggestions on how to make this scenario easier to use for the end user.

I'm trying to have a preset range for "any" which is gonna be the default for my application. I've read this old pull request comments (https://github.com/dangrossman/bootstrap-daterangepicker/pull/287) and they decided to forget about null dates and just set a big range instead, which is fine by me and how I originally saw it.

```
"ranges": {
	"Any": [
		moment("2000-01-01 00:00:00");
		moment().endOf('day');
	],
        [...other ranges here...]
}
```

My problem is that when any is selected and the user go to "Custom Range", the years in the left dropdown are preset to 2000, which is obviously not optimal since the user most likely want to search in the current year (even the current month). I would love that if the user goes from Any to Custom Range, that it switch the dates to becomes "today" automatically before opening the calendars, so the user don't have to change 2000 to 2017 manually.

Another problem is that, even if the user goes from "Any" and select "Today", and then go to Custom Range, it's still not working. Only the right calendar switch to the current date, while the left one stay to 2000. I debugged to code up to updateMonthsInView() and it see that both start/end dates are available in the right calendar, so it don't bother refreshing the left one. If instead of today I select a range that overlap two months, then both calendars refresh correctly.

And finally, there was an issue with the dropdown years displayed, which was cropped to -50,+5 from the original dates set, but I managed to fix it by setting minDate/maxDate values.

In other words, the "2000" should be used only as a result for the any date, but should not be displayed to the user anywhere.

**Comment by nhutminh1993** [comment]
I'm trying to find such a solution....please :(

**Comment by dangrossman** [comment]
The name of the predefined range clicked is the 3rd parameter passed to your callback function. That means you don't need to use any special dates to represent "any". You can check if "any" is the name of the label that was clicked, and react accordingly for your application, while having "any" set dates in the current year on the calendars.

**Comment by vinayo** [comment]
"ranges": {
	"Any": [
		moment("2000-01-01 00:00:00");
		moment().endOf('day');
	],
        [...other ranges here...]
}
 
please tell me how to set any as a default in this date range picker

**Comment by dangrossman** [comment]
To set "any" as the default, set the same dates as the `startDate` and `endDate` settings which define the initially selected range.

````
startDate: moment("2000-01-01 00:00:00"),
endDate: moment().endOf('day')
````
